packages/wallet/src/services/DelegationTracker/DelegationTracker.ts

```diff
@@ -68,6 +90,92 @@ export const certificateTransactionsWithEpochs = (
     )
   );
 
+export const createDelegationPortfolioTracker = ({
+  rewardAccounts$,
+  knownAddresses$,
+  utxoTracker
+}: DelegationPortfolioTrackerProps): Observable<Cip17Pool[]> => {
+  const balanceAllUtxosTracker = createBalanceByAddressTracker(utxoTracker);
+
+  const delegatedAccounts$ = rewardAccounts$.pipe(
+    map((rewardsAccounts) =>
+      rewardsAccounts.filter(
+        (account) => account.keyStatus === Cardano.StakeKeyStatus.Registered && account.delegatee?.nextNextEpoch
+      )
+    )
+  );
+
+  const hydratedDelegatedAccounts$ = delegatedAccounts$.pipe(
+    withLatestFrom(knownAddresses$),
+    map(([delegatedAccounts, knownAddresses]) =>
+      delegatedAccounts.map((delegatedAccount) => {
+        const groupedAddress = knownAddresses.find(
+          (knownAddr) => knownAddr.rewardAccount === delegatedAccount.address
+        )!;
+        return {
+          balance: createBalanceByAddressTracker(utxoTracker, groupedAddress.address),
+          delegatedAccount,
+          groupedAddress
+        };
+      })
+    )
+  );
+
+  return hydratedDelegatedAccounts$.pipe(
+    switchMap((accts) =>
+      iif(
+        () => accts.length === 0,
+        of([]),
+        combineLatest(
+          accts.map((acct) => {
+            const {
+              balance: {
+                utxo: { total$ }
+              },
+              delegatedAccount: { delegatee }
+            } = acct;
+            const pool = delegatee!.nextNextEpoch!;
+            return total$.pipe(
+              map(
+                (perAccountTotal): Cip17Pool => ({
+                  coins: perAccountTotal.coins,
+                  id: pool.id,
+                  name: pool.metadata?.name,
+                  percentage: 0, // rounding after aggregation to minimize precision loss
+                  pool,
+                  ticker: pool.metadata?.ticker
+                })
+              )
+            );
+          })
+        )
+      )
+    ),
+    // Merge rewardAccounts delegating to the same pool
+    map((cip17Pools) =>
+      _map(
+        _groupBy(cip17Pools, ({ id }) => id),
+        (pools): Cip17Pool =>
+          pools.reduce((mergedPool, pool) => ({
+            ...pool,
+            coins: pool.coins + mergedPool.coins || 0n,
+            percentage: pool.percentage + mergedPool.percentage || 0
```

Comment:
I just realized i should delete this line. Percentages are calculated in the next step

Reply:
[4a1e70c](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/4a1e70c437f0f710b0f098e7831f52bfb542c58c)